control/utils/graph.py

```
@@ -0,0 +1,327 @@
+            print(f"✔️ Successfully processed {len(timestamps)} data points from '{base_name}'.")
+
+            plot_and_save_rpm(
+                timestamps, rpms_data, args.angular_resolution_segments,
+                input_file, output_graph_filepath,
+                start_time=args.start_time, end_time=args.end_time
+            )
+            print(f"🖼️ Plot saved to '{output_graph_filepath}'.") # Changed emoji
+            num_graphs_generated += 1
+
+            try:
+                with open(output_log_filepath, 'w') as rpm_f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)